tests/scripts/thread-cert/Cert_6_1_02_REEDAttach.py

```diff
@@ -0,0 +1,377 @@
+            #         keep-alive message
+            #         The DUT MUST send MLE Child Update messages containing
+            #         the following TLVs:
+            #             - Leader Data TLV
+            #             - Mode TLV
+            #             - Source Address TLV
+            pkts.filter_wpan_src64(DUT).\
+                filter_wpan_dst64(REED).\
+                filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+                filter(lambda p: {
+                                  SOURCE_ADDRESS_TLV,
+                                  MODE_TLV,
+                                  LEADER_DATA_TLV
+                                 } < set(p.mle.tlv.type)
+                       ).\
+                must_next()
+
+            # Step 9: REED Respond to Child Update messages with a MLE Update
+            #         Response.
+            pkts.filter_wpan_src64(REED).\
+                filter_wpan_dst64(DUT).\
+                filter_mle_cmd(MLE_CHILD_UPDATE_RESPONSE).\
+                filter(lambda p: {
+                                  SOURCE_ADDRESS_TLV,
+                                  MODE_TLV,
+                                  LEADER_DATA_TLV
+                                 } < set(p.mle.tlv.type)
+                       ).\
+                must_next()
+
+            # Step 10: Leader Verifies connectivity by sending an ICMPv6 Echo Request
+            #          to the DUT link local address
+            #          The End Device MUST respond with ICMPv6 Echo Reply
+            _pkt = pkts.filter_ping_request().\
+                filter_ipv6_src_dst(REED_LLA, DUT_LLA).\
+                must_next()
+            pkts.filter_ping_reply(identifier=_pkt.icmpv6.echo.identifier).\
+                filter_ipv6_src_dst(DUT_LLA, REED_LLA).\
+                must_next()
+
+        else:
+            # Step 11: DUT sends periodic 802.15.4 Data Request messages as part
+            #          of the keep-alive message
+            #          The DUT must send a 802.15.4 Data Request command to the
+            #          parent device and receive an ACK message in response
+            _pkt2 = pkts.filter_wpan_src64(DUT).\
+                filter_wpan_dst16(_pkt.mle.tlv.source_addr).\
+                filter_wpan_cmd(WPAN_DATA_REQUEST).\
+                must_next()
+
+            pkts.filter(lambda p: p.wpan.seq_no == _pkt2.wpan.seq_no and\
+                        p.wpan.frame_type == 2).\
+                must_next()
+
+            # Step 12: REED verifies connectivity by sending an ICMPv6 Echo Request
+            #          to the DUT link local address
+            #          DUT responds with ICMPv6 Echo Reply
+
+            _pkt = pkts.filter_ping_request().\
+                filter_ipv6_src_dst(REED_LLA, DUT_LLA).\
+                must_next()
+            pkts.filter_ping_reply(identifier=_pkt.icmpv6.echo.identifier).\
+                filter_ipv6_src_dst(DUT_LLA, REED_LLA).\
+                must_next()
```

Comment:
If ping is from REED to DUT via link local address for step12 and step 10, there is similarity in the check, how about moving them out of the `if...else...` and only keeping one copy?

Reply:
yes, keep one